### PR TITLE
Add faster Cauchy

### PR DESCRIPTION
**Pull request by rohanbabbar04**
## Description
- Add `Cauchy` with all methods
- Add test in `test_scipy.py`.
- Add Cauchy to docs

## Checklist

- [x] Code style is correct (follows pylint and black guidelines)
- [x] Includes new or updated tests to cover the new feature
- [x] New features are properly documented (with an example if appropriate)

### CODE REVIEW

File: preliz/distributions/cauchy.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    def skewness(self):
+        return np.nan
+
+    def kurtosis(self):
+        return np.nan
+
+    def rvs(self, size=None, random_state=None):
+        random_state = np.random.default_rng(random_state)
+        random_samples = random_state.uniform(0, 1, size)
+        return nb_rvs(random_samples, self.alpha, self.beta)
```
Comment (aloctavodia) [inline preliz/distributions/cauchy.py:125]: The standard cauchy is available from numpy you just need to scale it and shift it https://numpy.org/doc/stable/reference/random/generated/numpy.random.standard_cauchy.html

File: preliz/distributions/cauchy.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+
+@nb.njit(cache=True)
+def nb_logpdf(x, alpha, beta):
+    return -np.log(np.pi) - np.log(beta) - np.log(1 + ((x - alpha) / beta) ** 2)
+
+
+@nb.njit(cache=True)
+def nb_neg_logpdf(x, mu, s):
+    return -(nb_logpdf(x, mu, s)).sum()
```
Comment (aloctavodia) [inline preliz/distributions/cauchy.py:157]:
```suggestion
@nb.njit(cache=True)
def nb_neg_logpdf(x, alpha, beta):
    return -(nb_logpdf(x, alpha, beta)).sum()
```

File: preliz/distributions/cauchy.py (hunk 1 above)
Comment (rohanbabbar04) [reply preliz/distributions/cauchy.py:125]: For this i did
`self.alpha + self.beta * random_state.standard_cauchy(size)`

but my answer was not matching, can you please correct me here

Comment (aloctavodia) [reply preliz/distributions/cauchy.py:125]: this is right. Maybe results do not match at the individual level, i.e. you are not getting exactly the same sequence of samples, the distributions should match. Anyway, tests are going to be simple if you keep this unchanged.

**Review by aloctavodia — APPROVED** [review]

**Pull request merged**